pkg/util/purge/resourcegroups.go

```diff
@@ -62,35 +64,54 @@ func (rc *ResourceCleaner) cleanResourceGroup(ctx context.Context, resourceGroup
 
 // cleanNetworking lists subnets in vnets and unnassign security groups
 func (rc *ResourceCleaner) cleanNetworking(ctx context.Context, resourceGroup mgmtfeatures.ResourceGroup) error {
-	secGroups, err := rc.securitygroupscli.List(ctx, *resourceGroup.Name, nil)
+	netwSecurityGroups, err := rc.securitygroupscli.List(ctx, *resourceGroup.Name, nil)
 	if err != nil {
 		return err
 	}
 
-	for _, secGroup := range secGroups {
-		if secGroup.Properties == nil || secGroup.Properties.Subnets == nil {
+	for _, networkSecGroup := range netwSecurityGroups {
+		if networkSecGroup.Properties == nil || networkSecGroup.Properties.Subnets == nil {
 			continue
 		}
 
-		for _, secGroupSubnet := range secGroup.Properties.Subnets {
-			subnet, err := rc.subnet.Get(ctx, *secGroupSubnet.ID)
+		for _, nsgSubnet := range networkSecGroup.Properties.Subnets {
+
+			vnetID, subnetName, err := apisubnet.Split(*nsgSubnet.ID)
+			if err != nil {
+				return err
+			}
+
+			vnetName, err := azure.ParseResourceID(vnetID)
+			if err != nil {
+				return err
+			}
+
+			subnetRGName, err := apisubnet.SplitRG(*nsgSubnet.ID)
+			if err != nil {
+				return err
+			}
+
+			subnet, err := rc.subnet.Get(ctx, subnetRGName, vnetName.ResourceName, subnetName, nil)
```

Comment:
You don't need to parse subnet ID one by one.
You can parse it like this.
https://github.com/Azure/ARO-RP/blob/88f29dee662395c5a34490030c7e15d3477aec3c/pkg/cluster/ensureendpoints.go#L33-L37

Reply:
Implemented, thank you!